docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc

```diff
@@ -0,0 +1,80 @@
+[[ml-nlp-deploy-models]]
+= Import and deploy trained models
+
+:keywords: {ml-init}, {stack}, {nlp}
+:description:  To make trained models available for inference, you must import \
+and deploy them in {es}.
+
+If you want to perform {nlp} tasks in your cluster, you must import and deploy
+an appropriate trained model. There are {es} APIs and tooling support in eland
+to prepare and manage models. 
+
+. <<ml-nlp-select-model,Choose a trained model.>>
+. <<ml-nlp-upload-model,Upload the trained model and vocabulary.>>
+. <<ml-nlp-deploy-model,Deploy the model in your cluster.>>
+
+[discrete]
+[[ml-nlp-select-model]]
+== Select a trained model
+
+Per <<ml-nlp-overview>>, there are multiple ways that you can use NLP features
+within the {stack}. After you determine which type of NLP task you want to
+perform, you must choose an appropriate trained model. You can either create the
+model yourself or find one that meets your needs.
+
+IMPORTANT: The {stack-ml-features} support transformer models that conform to
+the standard BERT model interface and use the WordPiece tokenization algorithm.
+
+The simplest method is to use a model that has already been fine-tuned for the
+type of analysis that you want to perform. For example, there are models and 
+data sets available for specific NLP tasks in
+https://huggingface.co/models[Hugging Face].
+
+[discrete]
+[[ml-nlp-upload-model]]
+== Upload the trained model and vocabulary
+
+After you find or create a model, you must upload it and its tokenizer
+vocabulary to your cluster. 
+
+Trained models must be in a TorchScript representation for use with
+{stack-ml-features}. To use scripts that format and upload the necessary
+artifacts, refer to https://github.com/elastic/eland#nlp-with-pytorch
+
+Alternatively, you can use the
+{ref}/put-trained-model-vocabulary.html[create trained model vocabulary API] and
+{ref}/put-trained-models.html[create trained models API] or
+{ref}/put-trained-model-definition-part.html[create trained model definition part API].
+// When you upload the model, it must be chunked and uploaded one chunk at a time. 
+//TBD Why? How?
+//Since eland encapsulates this process in a single Python method, it is the recommended method.
+
+[discrete]
+[[ml-nlp-deploy-model]]
+== Deploy the model in your cluster
+
+After you upload the model and vocabulary, you can use {kib} to view and manage
+their deployment across your cluster. Alternatively, you can use the
+{ref}/start-trained-model-deployment.html[start trained model deployment API].
+
+When you deploy the model, it is loaded on all available {ml} nodes. You can
+optionally specify the number of CPU cores it has access to on each node. In
+general, the total size of threading settings across all models on a node should
```

Comment:
We need to reference the API that allows you to set the thread settings. There are also two kinds of threads to set. In the original doc, I think I reference the two types `model_thread` and `inference_threads`. It's really important that we explain the difference between these two settings, either here or in the API.

Reply:
Yes, I played with adding that content but it felt too low-level for the current state of this page.  I'll open a PR to add it to the API for now and we can revisit as we get more examples into these docs.

Reply:
I changed my mind and added some of those recommendations in 23b14318e4492cb09b4a848deae1a7d7e2d55937

Reply:
I have, however, made some other minor edits in https://github.com/elastic/elasticsearch/pull/81549